packages/@n8n/nodes-langchain/nodes/agents/Agent/agents/ToolsAgent/execute.ts

```diff
@@ -346,11 +346,20 @@ export async function prepareMessages(
 		outputParser?: N8nOutputParser;
 	},
 ): Promise<BaseMessagePromptTemplateLike[]> {
-	const messages: BaseMessagePromptTemplateLike[] = [
-		['system', `{system_message}${options.outputParser ? '\n\n{formatting_instructions}' : ''}`],
-		['placeholder', '{chat_history}'],
-		['human', '{input}'],
-	];
+	const useSystemMessage = options.systemMessage ?? ctx.getNode().typeVersion < 1.9;
+
+	const messages: BaseMessagePromptTemplateLike[] = [];
+
+	if (useSystemMessage) {
+		messages.push([
+			'system',
+			`{system_message}${options.outputParser ? '{formatting_instructions}' : ''}`,
```

Comment:
Let's keep the new-lines from the original version here. Otherwise the formatting_instructions are merged together with the system message:
![CleanShot 2025-04-07 at 15 15 30](https://github.com/user-attachments/assets/7ce79aad-e321-4793-b708-858cd8519ffd)